libraries/AdaptiveExpressions/FunctionUtils.cs

```diff
@@ -1034,6 +1065,57 @@ internal static (object, string) NormalizeToDateTime(object timestamp, Func<Date
             return (result, error);
         }
 
+        internal static (CultureInfo, string) TryParseLocale(string localeStr)
+        {
+            CultureInfo result = null;
+            string error = null;
+            try
+            {
+                result = new CultureInfo(localeStr);
+            }
+            catch (Exception e)
+            {
+                error = e.Message;
+            }
+
+            return (result, error);
+        }
+
+        internal static (string, CultureInfo, string) DetermineFormatAndLocale(IReadOnlyList<object> args, string format, CultureInfo locale, int maxArgsLength)
+        {
+            string error = null;
+            if (maxArgsLength >= 2)
+            {
+                if (args.Count == maxArgsLength)
+                {
+                    // if the number of args equals to the maxArgsLength, the second last one is format, and the last one is locale
+                    format = args[maxArgsLength - 2] as string;
+                    (locale, error) = TryParseLocale(args[maxArgsLength - 1] as string);
+                }
+                else if (args.Count == maxArgsLength - 1)
+                {
+                    // if the number of args equals to the maxArgsLength - 1, the last one is format,
+                    format = args[maxArgsLength - 2] as string;
+                }
+            }
+
+            return (format, locale, error);
+        }
+
+        internal static (CultureInfo, string) DetermineLocale(IReadOnlyList<object> args, CultureInfo locale, int maxArgsLength)
+        {
+            string error = null;
+            if (maxArgsLength >= 2)
+            {
+                if (args.Count == maxArgsLength)
+                {
+                    (locale, error) = TryParseLocale(args[maxArgsLength - 1] as string);
+                }
```

Comment:
Should verify is a string and otherwise have an error with the value.  Otherwise this will end up as null.

Reply:
add verify string, if not update error message